epoch_publisher/src/client.rs

```diff
@@ -156,4 +208,45 @@ impl EpochPublisherClient {
         fbb.finish(fbb_root, None);
         fbb.finished_data()
     }
+
+    async fn background_loop(client: Arc<Self>, cancellation_token: CancellationToken) {
+        loop {
+            if cancellation_token.is_cancelled() {
+                return;
+            }
+
+            loop {
+                let expired_req_id = {
+                    let mut timeout_queue = client.timeout_queue.lock().await;
+                    if let Some(item) = timeout_queue.pop() {
+                        let (expiration_time, req_id) = item;
+                        if expiration_time < chrono::Utc::now() {
+                            req_id
+                        } else {
+                            break;
```

Comment:
Don't you need to put `item` back in the queue here or alternatively, use `peek` instead of `pop`?

Reply:
of course ...